src/Kiota.Builder/Writers/TypeScript/CodeUsingWriter.cs

```diff
@@ -18,18 +19,32 @@ public void WriteCodeElement(IEnumerable<CodeUsing> usings, CodeNamespace parent
         var externalImportSymbolsAndPaths = enumeratedUsings
                                                 .Where(static x => x.IsExternal)
-                                                .Select(static x => (x.Name, string.Empty, x.Declaration?.Name ?? string.Empty));
+                                                .Select(static x => new { Symbol = x.Name, Alias = string.Empty, Path = x.Declaration?.Name ?? string.Empty, ShouldUseTypeImport = false });
         var internalImportSymbolsAndPaths = enumeratedUsings
                                                 .Where(static x => !x.IsExternal)
-                                                .Select(x => _relativeImportManager.GetRelativeImportPathForUsing(x, parentNamespace));
-        var importSymbolsAndPaths = externalImportSymbolsAndPaths.Union(internalImportSymbolsAndPaths)
-                                                                .GroupBy(static x => x.Item3)
-                                                                .OrderBy(static x => x.Key);
+                                                .Select(x => new { CodeUsingPathTokens = _relativeImportManager.GetRelativeImportPathForUsing(x, parentNamespace), ShouldUseTypeImport = GetShouldUseTypeImport(x) });
+        var importSymbolsAndPaths = externalImportSymbolsAndPaths
+                                                .Union(internalImportSymbolsAndPaths.Select(static x => new { Symbol = x.CodeUsingPathTokens.Item1, Alias = x.CodeUsingPathTokens.Item2, Path = x.CodeUsingPathTokens.Item3, x.ShouldUseTypeImport }))
+                                                .GroupBy(x => x.Path)
+                                                .OrderBy(x => x.Key);
         foreach (var codeUsing in importSymbolsAndPaths.Where(static x => !string.IsNullOrWhiteSpace(x.Key)))
-            writer.WriteLine($"import {{{codeUsing.Select(static x => GetAliasedSymbol(x.Item1, x.Item2)).Distinct().OrderBy(static x => x).Aggregate(static (x, y) => x + ", " + y)}}} from '{codeUsing.Key}';");
+            writer.WriteLine($"import {codeUsing.Select(x => x.ShouldUseTypeImport ? "type " : "").Distinct().OrderBy(static x => x).Aggregate(static (x, y) => x)}" +
+                $"{{{codeUsing.Select(static x => GetAliasedSymbol(x.Symbol, x.Alias)).Distinct().OrderBy(static x => x).Aggregate(static (x, y) => x + ", " + y)}}} from '{codeUsing.Key}';");
 
         writer.WriteLine();
     }
+
+    /**
+    * Determines whether the import clause on TypeScript should include `import type` or just normal `import` statement
+    *
+    * @param codeUsing The code using statement to check
+    **/
+    private static bool GetShouldUseTypeImport(CodeUsing codeUsing)
+    {
+        // Check if codeUsing.Declaration is an instance of CodeType and if codeType.TypeDefinition is an instance of CodeInterface
+        return codeUsing.Declaration is CodeType codeType && codeType.TypeDefinition is CodeInterface;
```

Comment:
this is a good first step but it's not going to work for types we import from kiota abstractions that are interfaces

Reply:
disregard this comment, I've caught up on the different issues you've created now.

Reply:
I broke it down into parts, I wasn't sure what criteria I can use to distinguish whether an import from the non-generated code is an interface. https://github.com/microsoft/kiota/issues/2995